lib/get-config.js

```diff
@@ -322,7 +323,13 @@ module.exports = function(options) {
     processExtends(config, options);
     processIgnores(config);
 
-    validateRules(config, options);
+    validateRules(config.rules, config.loadedRules, options);
+
+    if (config.overrides) {
+      for (const override of config.overrides) {
+        validateRules(override.rules, config.loadedRules, options);
+      }
+    }
```

Comment:
Overrides should be able to do _basically_ anything that is done above. Some things that seem missing:

* extends
* pending
* ignore
* specifying plugins
* specifying rule configuration

I think it would _probably_ be fine to prevent `extends` and `plugins` for a first pass, but we'd need to give a nice error if overrides contained anything other than `rules` property for now.

Reply:
agreed... this first pass is just for the overrides for `rules`. I will add support for others in the next PR

Reply:
Sounds good, but it is very important that we add an error if you add "random properties" other than `rules` (otherwise we loose the ability to add new properties in the future without it being a breaking change).